puresec_cli/utils.py

```diff
@@ -193,8 +193,8 @@ def snakecase(string):
     """
 
     string = SNAKECASE_PATTERN1.sub(r"\g<1>_\g<2>", string)
-    string = SNAKECASE_PATTERN2.sub(r"\1_\2", string).lower()
-    return string
+    string = SNAKECASE_PATTERN2.sub(r"\1_\2", string)
+    return string.lower()
 
```

Comment:
Since it's utils and the target of the function is clear, you can maybe do it shorter
```
return SNAKECASE_PATTERN2.sub(r"\1_\2", SNAKECASE_PATTERN1.sub(r"\g<1>_\g<2>", string)).lower()
```

Reply:
If it was an object oriented language, I would do:

```python
return string.replace(...).replace(...).lower()
```

Unfortunately the creators of Python don't understand OOP... I don't like nesting functions the way you described (potentially infinite nesting with this style).